Engine/system/api/x11api.cpp

```diff
@@ -197,19 +224,72 @@ void X11Api::implDestroyWindow(SystemApi::Window *w) {
   }
 
 bool X11Api::implSetAsFullscreen(SystemApi::Window *w, bool fullScreen) {
-  return false;
+  Atom a[1];
+  a[0] = _NET_WM_STATE_FULLSCREEN();
+
+  if(fullScreen) {
+    XChangeProperty ( dpy, HWND(w), _NET_WM_STATE(),
+      XA_ATOM, 32, PropModeReplace, (unsigned char*)a, 1);
+  } else {
+    XChangeProperty ( dpy, HWND(w), _NET_WM_STATE(),
+      XA_ATOM, 32, PropModeReplace, (unsigned char*)a, 0);
+  }
+  
+  XSync(dpy,False);
   }
 
 bool X11Api::implIsFullscreen(SystemApi::Window *w) {
+  Atom actualType;
+  int actualFormat;
+  unsigned long i, numItems, bytesAfter;
+  unsigned char *propertyValue = NULL;
+  long maxLength = 1024;
+  
+  if (XGetWindowProperty(dpy, HWND(w), _NET_WM_STATE(),
+                           0l, maxLength, False, XA_ATOM, &actualType,
+                           &actualFormat, &numItems, &bytesAfter,
+                           &propertyValue) == Success) {
+        int fullscreen = 0;
+        Atom *atoms = (Atom *) propertyValue;
+
+        for (i = 0; i < numItems; ++i) {
+            if (atoms[i] == _NET_WM_STATE_FULLSCREEN()) {
+                fullscreen = 1;
+            }
+        }
+        
+        if(fullscreen) {
+          return true;
+        }
+  }
   return false;
   }
 
 void X11Api::implSetCursorPosition(int x, int y) {
-  // TODO
+  activeCursorChange = 1;
+  XWarpPointer(dpy, None, HWND(windows.begin()->first), 0, 0, 0, 0, x, y);
```

Comment:
Hm, I don't like unconditional access to ::windows.begin().
Is it possible to use here `XRootWindow` instead?
```c++
// pseudo code
auto root = XRootWindow(dpy, 0);
XWarpPointer(dpy, None, root, 0, 0, 0, 0, x, y);
```